bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go

```diff
@@ -63,6 +63,23 @@ const (
 	KubeadmConfigControllerName = "kubeadmconfig-controller"
 )
 
+var (
+	// controlPlaneTaint is the taint that kubeadm applies to the control plane nodes.
+	// The values are copied from kubeadm codebase.
```

Comment:
Let's add a similar note here that this taint is only added on >= 1.24 (xref: https://github.com/kubernetes-sigs/cluster-api/issues/3279#issuecomment-1271598300)